test/endToEnd/addModelToAnalysisTest.js

```diff
@@ -0,0 +1,294 @@
+    .waitForElementVisible('#convergence-diagnostics-table')
+    .waitForElementVisible('#relative-effects-table')
+    .waitForElementVisible('#relative-effects-plots')
+    .waitForElementVisible('#rank-probabilities-plot')
+    .waitForElementVisible('#rank-probabilities-table')
+    .waitForElementVisible('#absolute-residual-deviance-table')
+    .click('#logo');
+}
+
+function verifyLeaveOneOutModelContents(browser) {
+  return verifyCommonContent(browser)
+    .assert.containsText('#model-sub-type', 'leave one out')
+    .waitForElementVisible('#convergence-diagnostics-table')
+    .waitForElementVisible('#random-effects-standard-deviation')
+    .waitForElementVisible('#absolute-treatment-effects')
+    .waitForElementVisible('#study-effect-funnel-plot')
+    .waitForElementVisible('#relative-effects-table')
+    .waitForElementVisible('#relative-effects-plots')
+    .waitForElementVisible('#absolute-residual-deviance-table');
+}
+
+function verifyDesignAdjustedModelContents(browser) {
+  return verifyCommonContent(browser)
+    .assert.containsText('#model-likelihood', 'normal / identity')
+    .waitForElementVisible('#convergence-diagnostics-table')
+    .waitForElementVisible('#random-effects-standard-deviation')
+    .waitForElementVisible('#absolute-treatment-effects')
+    .waitForElementVisible('#study-effect-funnel-plot')
+    .waitForElementVisible('#relative-effects-table')
+    .waitForElementVisible('#relative-effects-plots')
+    .waitForElementVisible('#rank-probabilities-plot')
+    .waitForElementVisible('#rank-probabilities-table')
+    .waitForElementVisible('#absolute-residual-deviance-table');
+}
+
+function verifyCommonContent(browser, modelType = 'network', modelEffectsType = 'random') {
+  return browser
+    .assert.containsText('#model-label', MODEL_TITLE)
+    .assert.containsText('#model-view-analysis', TITLE)
+    .assert.containsText('#model-view-outcome', OUTCOME)
+    .assert.containsText('#model-type', modelType)
+    .assert.containsText('#model-effects-type', modelEffectsType)
+    .waitForElementVisible('#residual-deviance-plot')
+    .waitForElementVisible('#model-fit-table')
+    .waitForElementVisible('#rank-probabilities');
+}
+
+function beforeEach(browser) {
+  browser.resizeWindow(1366, 728);
+  loginService.login(browser)
+    .waitForElementVisible('#analyses-header');
+}
+
+function afterEach(browser) {
+  analysesService.deleteFromList(browser).end();
+}
+
+function fixedEffectSpecificPairwiseModel(browser) {
+  const modelSettings = {
+    title: MODEL_TITLE,
+    effectsType: FIXED_EFFECT_MODEL,
+    modelMainType: '#pairwise-model-main-type-radio',
+    modelSubType: '#pairwise-specific-type-radio'
+  };
+  const pairwiseComparison = 'Fluoxetine — Paroxetine';
+
+  analysesService.addAnalysis(browser, TITLE, OUTCOME, '/example.json');
+  modelService.addModel(browser, modelSettings);
+  verifyCommonContent(browser, 'pairwise', 'fixed')
```

Comment:
Very neat. Unfortunately hard to check actual values due to stochasticity, so it's good to at least check the ranking.

Reply:
Shouldn't there be an R unit-test for this?